diff --git a/rspamd.service b/rspamd.service
index 1d1180861..5a57d2dfa 100644
--- a/rspamd.service
+++ b/rspamd.service
@@ -8,9 +8,12 @@ LimitNOFILE=1048576
 NonBlocking=true
 ExecStart=/usr/bin/rspamd -c /etc/rspamd/rspamd.conf -f
 ExecReload=/bin/kill -HUP $MAINPID
-User=_rspamd
+User=rspamd
+Group=rspamd
 RuntimeDirectory=rspamd
 RuntimeDirectoryMode=0755
+StateDirectory=rspamd
+LogsDirectory=rspamd
 Restart=always
 
 [Install]
